pkg/controller/certificates/clustertrustbundlepublisher/publisher_test.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustertrustbundlepublisher
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	cryptorand "crypto/rand"
+	"testing"
+	"time"
+
+	certificatesv1alpha1 "k8s.io/api/certificates/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+	"k8s.io/client-go/kubernetes/fake"
+	certalpha1listers "k8s.io/client-go/listers/certificates/v1alpha1"
+	clienttesting "k8s.io/client-go/testing"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/kubernetes/test/utils/ktesting"
+)
+
+const testSignerName = "test.test/testSigner"
+
+func TestCTBPublisherSync(t *testing.T) {
+	checkCreatedTestSignerBundle := func(t *testing.T, actions []clienttesting.Action) {
+		filteredActions := filterOutListWatch(actions)
+		if len(filteredActions) != 1 {
```

Comment:
Its fine if you do this in the integration test, but I would like to see something that exercises when the sync loop does both the `create/update` bit while also doing the `delete` bit in a single loop iteration.

Reply:
addressed in the unit tests